### PR TITLE
Add Git pre-commit and pre-push hooks to Gradle build

### PR DESCRIPTION
**Pull request by Aulud**
## Add Git pre-commit and pre-push hooks to Gradle build
Summary: **This PR is experimental and may break development workflow.** Introduces Git hooks into the development workflow to prevent failed builds on Travis.

### Changelog
- Amend Gradle build file to copy Git hook scripts to `./.git/hooks`
   - `pre-commit` and `pre-push` scripts will be called respectively prior to making a local commit or pushing to the remote repo
   - A commit or push will only succeed if the script exits without error

Last updated 15 Oct 2019, 7:00PM

### CODE REVIEW

**Comment by Aulud** [comment]
You may have to **manually trigger Gradle once** for the hooks to be copied into the `./.git/hooks` folder. This requires Bash (Git Bash will do).

@daekoon @podocarp Please test if this branch works on Ubuntu 18.04 LTS with Git CLI.

@liakify @JohnNzj Please test if this branch works on Windows with IntelliJ.

File: config/hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+./gradlew hooks checkstyleMain checkstyleTest asciidoctor
+
+# Store the last exit code in a variable
+RESULT=$?
+
+if [ $RESULT -eq 0 ]
+then
+    echo -e "\n================ Pre-commit checks: PASSED ================\n"
+else
+    echo -e "\n================ Pre-commit checks: FAILED ================"
```
Comment (podocarp) [inline config/hooks/pre-commit]: missing one \n at the end

File: config/hooks/pre-push [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+./gradlew hooks clean checkstyleMain checkstyleTest asciidoctor test
+
+# Store the last exit code in a variable
+RESULT=$?
+
+if [ $RESULT -eq 0 ]
+then
+    echo -e "\n================ Pre-push checks: PASSED ================\n"
+else
+    echo -e "\n================ Pre-push checks: FAILED ================"
```
Comment (podocarp) [inline config/hooks/pre-push]: missing another \n at the end

File: config/hooks/pre-commit (hunk 1 above)
Comment (Aulud) [reply config/hooks/pre-commit]: Oh dank thanks! Oddly, the output is a bit botched on Windows, but it's Windows, so...

**Review by liakify — APPROVED** [review]
yes

**Comment by Aulud** [comment]
Merged since there are no objections. Thanks @liakify!

**Pull request merged**